### ISSUE TITLE
SQL Server `.findMany()` with `include` fails to chunk out query if results are > 2100 rows

### ISSUE THREAD

**Issue by PatricioNG** [prisma/prisma#21778]
### Bug description

For SQL Server, when you're querying a table using `.findMany()` and use a `include` for querying a relation, Prisma will query the related table and insert a new parameter for every result returned by the initial `.findMany()`. This causes errors when your results for the initial table are greater than 2100 rows as SQL Server limits parameters to 2100 at a time. 

This behaviour does not occur for `createMany` statements, as those are properly chunked out to avoid hitting the 2100 parameter limit.

### How to reproduce

I have created a basic reproduction here https://github.com/PatricioNG/prisma-sql-server-parameter-error/tree/main

Steps are included in the README.md, but really this can be reproduced using any table that has a relation - if the base table has more than 2100 records, query it and include the related table's results using `include`. The DB will throw an error as Prisma attempts to send queries with more than 2100 parameters.

### Expected behavior

Prisma should chunk out the query of the related table and then reassemble the results before returning the response.

### Prisma information

```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "sqlserver"
  url      = env("DATABASE_URL")
}

model Authors {
  id        Int     @id @default(autoincrement())
  firstName String
  lastName  String
  Posts     Posts[]
}

model Posts {
  id          Int     @id @default(autoincrement())
  authorsId   Int
  title       String
  description String
  authors     Authors @relation(fields: [authorsId], references: [id])
}
```

```
import { PrismaClient } from '@prisma/client';
const prisma = new PrismaClient();

const SEED_ROW_COUNT = 2200;

const seedDB = async () => {
	let count = 1;

	const authorsInsertData = [];
	const postsInsertData = [];

	//Seed data
	while (count <= SEED_ROW_COUNT) {
		authorsInsertData.push({
			firstName: 'FIRST_NAME',
			lastName: 'LAST_NAME',
		});

		postsInsertData.push({
			authorsId: count,
			description: 'DESCRIPTION',
			title: 'TITLE',
		});
		count++;
	}

	await prisma.authors.createMany({
		data: authorsInsertData,
	});

	await prisma.posts.createMany({
		data: postsInsertData,
	});
};

const Main = async () => {
	//Seed command will insert 2200
	//records into the table - prisma handles this just
	//fine by splitting up the inserts
	await seedDB();

	//Attempting to pull back 2200 with an
	//`include` statement will throw an error
	//due to the amount of parameters - it will attempt to
	//send a query with 2200 parameters instead of respecting the
	//2100 limit on SQL Server
	await prisma.authors.findMany({
		include: {
			Posts: true,
		},
	});
};

Main();
```

### Environment & setup

- OS: Windows
- Database: SQL Server
- Node.js version: 20.3.1

### Prisma Version

```
"@prisma/client": "^5.5.2",
    "prisma": "^5.5.2"
```

**Comment by PatricioNG** [comment]
Sorry for the direct tag @janpio, just curious if there's a process to help provide any more information to get smaller less priority issues like this on the pipeline for a fix from my/the users side of things? 🙏 I'm sure it's low priority in the grand scheme of things but it does sadly make Prisma client harder to work with. Sadly, I don't have the knowledge to jump into the Rust side of things and help PR a fix, so wondering if there's any additional info gathering I can do?

Happened to add a few additional relations to the DB at our company and didn't realize this issue would come up, but it's now breaking quite a few areas, all good as we can work around it by manually adding in the chunking logic but it sort of defeats the purpose of using relations in Prisma. Easiest solve is to use `$queryRaw` anytime there's a relation but then that again just sort of defeats the purpose of Prisma 😞

**Comment by janpio** [comment]
Thanks for the excellent reproduction, I can fully confirm this:

```
PrismaClientUnknownRequestError:
Invalid `prisma.authors.findMany()` invocation in
c:\Users\Jan\Documents\throwaway\21778\run.ts:10:23

   7 //due to the amount of parameters - it will attempt to
   8 //send a query with 2200 parameters instead of respecting the
   9 //2100 limit on SQL Server
→ 10 await prisma.authors.findMany(
Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(TokenError { code: 8003, state: 1, class: 16, message: "The incoming request has too many parameters. The server supports a maximum of 2100 parameters. Reduce the number of parameters and resend the request.", server: "EC2AMAZ-O13R6ML", procedure: "", line: 1 }), transient: false })
    at _n.handleRequestError (C:\Users\Jan\Documents\throwaway\21778\node_modules\@prisma\client\runtime\library.js:123:7044)
    at _n.handleAndLogRequestError (C:\Users\Jan\Documents\throwaway\21778\node_modules\@prisma\client\runtime\library.js:123:6188)
    at _n.request (C:\Users\Jan\Documents\throwaway\21778\node_modules\@prisma\client\runtime\library.js:123:5896)
    at async l (C:\Users\Jan\Documents\throwaway\21778\node_modules\@prisma\client\runtime\library.js:128:10871)
    at Main (c:\Users\Jan\Documents\throwaway\21778\run.ts:10:2) {
  clientVersion: '5.10.1'
}
```

**Comment by janpio** [comment]
And now I am closing this as a duplicate of this older issue: https://github.com/prisma/prisma/issues/18203 We'll continue there.

**Comment by PatricioNG** [comment]
Thanks so much for the env variable tip on this one 👏 happy we have a workaround to save us the pain!

**Comment by janpio** [comment]
For anyone landing here via Google:

This is currently bug and not intentional.
Until that is fixed (follow #18203 for updates) you can set an environment variable `QUERY_BATCH_SIZE=2098` (or lower) as a workaround that might influence the chunking to work correctly and let you proceed.
Thanks.
